internal/controller/pooler_controller.go

```diff
@@ -156,6 +161,56 @@ func isOwnedByPooler(obj client.Object) (string, bool) {
 	return owner.Name, true
 }
 
+func isOwnedByPooler(poolerName string, obj client.Object) bool {
+	ownerName, isOwned := isOwnedByPoolerKind(obj)
```

Comment:
If you agree, I would refactor
`if !isOwned {
		return false
	}
	return poolerName == ownerName`
	
with one liner

`return isOwned && poolerName == ownerName`